pkg/osquery/runtime/osqueryinstance.go

```diff
@@ -219,10 +219,9 @@ type OsqueryInstance struct {
 	doneCtx                 context.Context
 	cancel                  context.CancelFunc
 	cmd                     *exec.Cmd
-	emsLock                 sync.RWMutex
+	emsLock                 sync.RWMutex // Lock for extensionManagerServers
```

Comment:
I'm not sure still need this. Maybe? I can't remember if the intent was to gate access to the thrift socket, which osquery-go should be doing, or gate behavior inside our extensions

Reply:
This is just a lock for the array so that we don't add/remove servers from the array while we're reading from it, so I think we do still want it. I added a comment about this below.

Reply:
Oh, I see. I guess that makes sense. I'm not sure our normal usage hits it, but sure.